subxt/examples/unstable_light_client.rs

```diff
@@ -0,0 +1,150 @@
+//! This is a comprehensive example that utilizes the Light Client
+//! for a subset of subxt requests.
+//!
+//! Includes:
+//!  - Subscribes to all finalized blocks using the old RPC method
+//!  - Subscribes to the head of the chain using the new `chainHead` RPC method
+//!  - Dynamically query constants
+//!  - Dynamically decode the events of the latest block
+//!  - Various RPC calls to ensure proper shape of the response.
+//!
+//! # Note
+//!
+//! This feature is experimental and things might break without notice.
+
+use futures::StreamExt;
+use std::sync::Arc;
+use subxt::{
+    rpc::{types::FollowEvent, LightClientBuilder},
+    OnlineClient, PolkadotConfig,
+};
+
+// Generate an interface that we can use from the node's metadata.
+#[subxt::subxt(runtime_metadata_path = "../artifacts/polkadot_metadata_small.scale")]
+pub mod polkadot {}
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    tracing_subscriber::fmt::init();
+
+    // Create a light client from the provided chain spec.
+    // Note: this connects to the live polkadot chain.
+    let light_client = LightClientBuilder::new()
+        .chain_spec(include_str!("../artifacts/polkadot_spec.json"))
```

Comment:
Is it a good idea to keep the `artifacts/polkadot_spec.json` around?

For documentation, we can just show an example that points to some made up spec path (since we `ignore` the docs pointing to this anyway). And then in the "full" example we can build from url anyway :))

Maybe it's worth keeping around still but I'm not sure if it'll end up falling out of date or something (or maybe it never would)? It's quite large (5.5MB) so I'm hesitant :)